modules/lang-mustache/src/main/java/org/elasticsearch/script/mustache/MustacheScriptEngine.java

```diff
@@ -47,6 +49,8 @@ public final class MustacheScriptEngine implements ScriptEngine {
 
     public static final String NAME = "mustache";
 
+    private static final int MUSTACHE_SIZE_LIMIT = 1024 * 1024 * 1024;  // 1GB
```

Comment:
Chosen fairly arbitrarily (the Java char array limit is 2GB). Should we make this dynamic based on heap size?

Reply:
1GB seems enormous. Scripting is limited to 65k by default (it's adjustable via a node setting). That's the source, but still, the rendered template shouldn't be _that_ much bigger than the source, right? I would cap this at 1MB, and if necessary we can make it configurable at a node level like the max script size.

Reply:
I've changed the default size limit to 1MB, and added a node setting